### ISSUE TITLE
Container using different memory than the server

### ISSUE THREAD

**Issue by JeanLucaMicheluzzi** [pterodactyl/panel#5239]
### Current Behavior

I was seeing that the server on the panel was showing very high memory usage, whereas using the server locally it was not using even half of what was being shown. 

I installed the server analysis plugin and noticed that it was showing a memory value, but on the panel it was twice as much. 

It's okay that we must take Docker's memory usage into consideration, but using more than 1GB is almost unfeasible. I see this behavior on all my servers that I have on the panel.

![Screenshot_65](https://github.com/user-attachments/assets/c292132c-c54d-4833-a4dd-c2eeb9d747c9)

### Expected Behavior

Validate the behavior of the Panel, I see that the same problem occurs on several hosting

### Steps to Reproduce

Start te server

### Panel Version

1.11.7

### Wings Version

1.11.13

### Games and/or Eggs Affected

Minecraft (Paper)

### Docker Image

ghcr.io/pterodactyl/yolks:java_16

### Is there an existing issue for this?

- [X] I have searched the existing issues before opening this issue.
- [X] I have provided all relevant details, including the specific game and Docker images I am using if this issue is related to running a server.
- [X] I have checked in the Discord server and believe this is a bug with the software, and not a configuration issue with my specific system.

**Comment by yesBad** [comment]
This is not really something feasible to "fix".

The panel (aka Docker stats) only show the allocated memory for the Java instance. It however doesn't mean it's "using" that much.

**Comment by JeanLucaMicheluzzi** [comment]
Ok, but shouldn't it just show the real value of usage by the server and discard the container's usage without accounting for its memory? Really leaving the total ram for the server.

We can take for example the panel with 1GB, I make 1GB available for the minecraft server, but in the middle of use it ends up reaching 100% of the memory usage not by the server alone, but by the docker + server causing the instance to be restarted.

**Comment by PLASMAchicken** [comment]
You do not understand how Memory works. Also the server does not get restarted when it reaches max, except if the process itself cannot work with that memory limit.
In Minecraft if you press F3 you can actually see it pretty good.
![image](https://github.com/user-attachments/assets/04f94f27-e506-4d42-b6fa-ddb9a64728e9)

So my client is using 1061MB currently, but it has Allocated 33% aka 1993MB, this is the usage that would show up in the panel.
But the max memory it can allocate is 5993MB.
 Especially when playing modded Minecraft its better to tell Java to allocate the whole Ram you assign to it from the start, this will look like this:
![javaw_spUUiV11C8](https://github.com/user-attachments/assets/2669439b-f5fc-4748-a37b-19ee058f3e1c)
Now Windows Task Manager / Pterodactyl will see 6GB reseved/in use, but spark health will show the 1318MB used.

**TLDR: You can acutally see the 2.62GB ALLOCATED in your spark health screen. There is a red marker at the 2.6GB position**

**Comment by danny6167** [comment]
The panel is working perfectly fine. It is reporting the amount of memory in use by the processes inside the container as it's designed to do.
The number you are looking at is the heap utilization inside java and is not the same as actual system memory utilization.
